transformers/commands/user.py

```diff
@@ -0,0 +1,122 @@
+from argparse import ArgumentParser
+from getpass import getpass
+import os
+
+from transformers.commands import BaseTransformersCLICommand
+from transformers.hf_api import HfApi, HfFolder, HTTPError
+
+
+class UserCommands(BaseTransformersCLICommand):
+    @staticmethod
+    def register_subcommand(parser: ArgumentParser):
+        login_parser = parser.add_parser('login')
+        login_parser.set_defaults(func=lambda args: LoginCommand(args))
+        whoami_parser = parser.add_parser('whoami')
+        whoami_parser.set_defaults(func=lambda args: WhoamiCommand(args))
+        logout_parser = parser.add_parser('logout')
+        logout_parser.set_defaults(func=lambda args: LogoutCommand(args))
+        list_parser = parser.add_parser('ls')
+        list_parser.set_defaults(func=lambda args: ListObjsCommand(args))
+        # upload
+        upload_parser = parser.add_parser('upload')
+        upload_parser.add_argument('file', type=str, help='Local filepath of the file to upload.')
+        upload_parser.add_argument('--filename', type=str, default=None, help='Optional: override object filename on S3.')
+        upload_parser.set_defaults(func=lambda args: UploadCommand(args))
```

Comment:
@mfuntowicz I hesitated between doing `n` subclasses of `BaseTransformersCLICommand` vs. grouping all of them in one `register_subcommand` call like here.

In the current state, `UserCommands` is not even a real `BaseTransformersCLICommand` (it doesn't have a `run()`) so I can maybe remove the reference to `BaseTransformersCLICommand `. What do you think?

Reply:
I'm fine keeping things this way. 

I want to stick to the original design, you may want to remove ```class UserCommand```, and merge it with ```class BaseUserCommand```:

```python
class BaseUserCommand(ABC, BaseTransformersCLICommand):
    def __init__(self, args):
        self.args = args
        self._api = HfApi()
``` 

And then put all the register_subcommand on the specific ```Command``` implementation. 

Keeping a common ancestor to all the commands might be interesting if we need to document / dynamically list all the commands at some point.

Reply:
Up to you really! And we can always revisit later of course.

Reply:
Let's go this way, I'll include the change when pushing the CLI next week